Assets/Scripts/MVC/PlayerTank/TankController.cs

```diff
@@ -2,27 +2,36 @@
 
 public class TankController
 {
+    public float bulletFireCount { get; private set; }
     public TankModel tankModel;
     public TankView tankView;
+    float distance;
+    bool distanceNotified;
     public TankController(TankView tankView, Joystick joystick, PlayerTankScriptableObject playerTankScriptableObject){
         this.tankModel = new TankModel(playerTankScriptableObject);
         this.tankView = GameObject.Instantiate<TankView>(tankView);
         this.tankView.setupJoyStick(joystick);
         this.tankView.setTankController(this);
+        bulletFireCount = 0;
+        distanceNotified = false;
     }
 
     public void tankMovement(float horizontal, float vertical){
         Vector3 currentPosition = this.tankView.transform.position;
-        if(Input.touchCount > 0){
+        // if(Input.touchCount > 0){
             currentPosition.x += (horizontal * Time.deltaTime) * this.tankModel.movementSpeed;
             currentPosition.z += (vertical * Time.deltaTime) * this.tankModel.movementSpeed;
-        }
+        // }
         this.tankView.transform.position = currentPosition;
     }
 
     public void fireBullet(Transform bulletSpawnPosition)
     {
+        bulletFireCount++;
```

Comment:
Move it to an achievement class or even better create multiple types of achievement classes and put it in there.

Reply:
bulletfirecount should be declared where it is supposed to be used.